pyproject.toml

```diff
@@ -1,16 +1,116 @@
-[tool.black]
-target-version = ['py38', 'py39', 'py310', 'py311']
-include = '\.pyi?$'
-exclude = '''
-/(
-    \.eggs
-  | \.git
-  | \.pytest_cache
-  | htmlcov
-  | figures
-  | build
-  | dist
-  # Specific to this package
-  | becquerel.egg-info
-)/
-'''
+
+# all of ruff's settings
+# https://docs.astral.sh/ruff/settings/
+# all of ruff's rules:
+# https://docs.astral.sh/ruff/rules/
+
+[tool.ruff]
+namespace-packages = ["becquerel"]
+target-version = "py38"
+fix = true
+show-fixes = true
+preview = false
+
+# format the same as black
+indent-width = 4
+line-length = 88
+
+[tool.ruff.format]
+indent-style = "space"
+quote-style = "double"
+line-ending = "auto"
+skip-magic-trailing-comma = false
+docstring-code-format = true
+docstring-code-line-length = "dynamic"
+
+[tool.ruff.lint]
+select = [
+  "F",       # pyflakes
+  "E",       # pycodestyle errors
+  "W",       # pycodestyle warnings
+  # "C",       # mccabe
```

Comment:
wouldn't mind turning mccabe on...

Reply:
Agreed, but we have a lot of code that has high complexity, and fixing these require more than just cosmetic changes to the code:

```
becquerel/core/autocal.py:112:5: C901 `find_best_gain` is too complex (16 > 10)
becquerel/core/calibration.py:35:5: C901 `_validate_domain_range` is too complex (11 > 10)
becquerel/core/calibration.py:182:5: C901 `_validate_expression` is too complex (17 > 10)
becquerel/core/calibration.py:418:5: C901 `_check_points` is too complex (12 > 10)
becquerel/core/calibration.py:898:9: C901 `read` is too complex (12 > 10)
becquerel/core/fitting.py:759:9: C901 `fit` is too complex (22 > 10)
becquerel/core/fitting.py:1227:9: C901 `custom_plot` is too complex (24 > 10)
becquerel/core/rebin.py:283:5: C901 `rebin` is too complex (15 > 10)
becquerel/core/spectrum.py:87:9: C901 `__init__` is too complex (17 > 10)
becquerel/core/spectrum.py:638:9: C901 `write` is too complex (14 > 10)
becquerel/parsers/cnf.py:80:5: C901 `read` is too complex (54 > 10)
becquerel/parsers/iec1455.py:34:5: C901 `read` is too complex (34 > 10)
becquerel/parsers/spc.py:183:5: C901 `read` is too complex (38 > 10)
becquerel/parsers/spe.py:13:5: C901 `read` is too complex (26 > 10)
becquerel/tools/isotope.py:17:5: C901 `_split_element_mass` is too complex (13 > 10)
becquerel/tools/materials.py:16:5: C901 `_load_and_compile_materials` is too complex (16 > 10)
becquerel/tools/nndc.py:107:5: C901 `_parse_headers` is too complex (15 > 10)
becquerel/tools/nndc.py:207:5: C901 `_parse_float_uncertainty` is too complex (21 > 10)
becquerel/tools/nndc.py:459:9: C901 `update` is too complex (13 > 10)
becquerel/tools/xcom.py:260:9: C901 `update` is too complex (17 > 10)
tests/isotope_test.py:32:5: C901 `test_isotope_init_args` is too complex (11 > 10)
```

Reply:
agreed, let's save this for the future. We could also increase the threshold to 20 to narrow that list down to 7